### DISCUSSION TITLE
MAC系统本地开发怎么调试？

### DISCUSSION THREAD

**Discussion by luyuanx2** [harmonycloud/kindling#602] [agent collector]
运行直接报错
<img width="1624" alt="image" src="https://github.com/KindlingProject/kindling/assets/26129033/f9bc09d7-b69a-4d7c-abe4-35fe69935162">

**Comment by dxsup** [comment]
项目基于Linux eBPF，只能运行在Linux系统上。要使用Mac进行调试，请尝试Docker或Linux虚拟机。

**Reply by luyuanx2** [reply]
请问使用docker，具体怎么操作呢

**Reply by dxsup** [reply]
参考 https://code.visualstudio.com/docs/devcontainers/containers

可使用Docker image: kindlingproject/agent-builder:latest

**Reply by luyuanx2** [reply]
容器内运行collector报错
![image](https://github.com/KindlingProject/kindling/assets/26129033/3940185c-9a70-4c73-8196-d48f2055e7b2)

**Reply by dxsup** [reply]
请查看日志中的内容。按照FAQ的说明进行操作。

项目依赖于eBPF，因此需要为内核编译eBPF probe，这个报错的意思是缺少该内核版本的eBPF probe，所以你需要自行编译，并在启动collector前通过环境变量`SYSDIG_BPF_PROBE`指定编译出来的`probe.o`的路径：
```shell
export SYSDIG_BPF_PROBE="/opt/probe.o"
```

**Reply by luyuanx2** [reply]
指定了probe文件，启动后还是有报错
<img width="1701" alt="image" src="https://github.com/KindlingProject/kindling/assets/26129033/72648ce5-0935-46b6-9b3c-4b7347f53171">

**Reply by dxsup** [reply]
启动时请使用`privileged`权限
